lib/query/index.js

```diff
@@ -0,0 +1,129 @@
+const _ = require('lodash')
+const { filterAndTransform } = require('./filter-and-transform')
+const { logWarnings } = require('./log-warnings')
+const { renderFeaturesResponse } = require('./render-features')
+const { renderStatisticsResponse } = require('./render-statistics')
+const { renderPrecalulatedStatisticsResponse } = require('./render-precalculated-statistics')
+const { renderCountAndExtentResponse } = require('./render-count-and-exent')
+const { getGeometryTypeFromGeojson } = require('../helpers')
+
+function query (json, requestParams = {}) {
+  const {
+    features,
+    filtersApplied: {
+      all: skipFiltering
+    } = {}
+  } = json
+
+  const { f: requestedFormat } = requestParams
+
+  if (shouldRenderPrecalculatedData(json, requestParams)) {
+    return renderPrecalculatedData(json, requestParams)
+  }
+
+  const data = (skipFiltering || !features) ? json : filterAndTransform(json, requestParams)
+
+  if (shouldLogWarnings()) {
+    logWarnings(data, requestParams.f)
+  }
+
+  if (requestedFormat === 'geojson') {
+    return {
+      type: 'FeatureCollection',
+      features: data.features
+    }
+  }
+
+  return renderGeoservicesResponse(data, {
+    ...requestParams,
+    attributeSample: _.get(json, 'features[0].properties'),
+    geometryType: getGeometryTypeFromGeojson(json)
+  })
+}
+
+function shouldRenderPrecalculatedData ({ statistics, count, extent }, { returnCountOnly, returnExtentOnly }) {
+  if (statistics) {
+    return true
+  }
+
+  if (returnCountOnly === true && count !== undefined && returnExtentOnly === true && extent) {
```

Comment:
Are all the strict equalities redundant here?

Reply:
Not sure what you mean by redundant.  Yes, they need to be strict.  Falsy or truthy values in these cases lead to problems.

Reply:
Wasn't sure if `returnCountOnly` and `returnExtentOnly` can be anything other than a boolean, but I think your last statement answers my question. I was just thinking of simplifying it to 

`(returnCountOnly && count !== undefined && returnExtentOnly && extent)`